Add new PR templates and revamp issue templates (#3711)
EssentialsX/Essentials: .github/PULL_REQUEST_TEMPLATE/new-feature.md

      recommend opening a new feature request BEFORE opening your PR to
      implement it, as this allows us to review whether we're likely to accept
      your feature in advance, and also allows us to discuss possible
      implementations for the feature. If there is no associated feature
      request, your PR may be delayed or rejected without warning.
      
      You can open a new feature request by following this link:
      https://github.com/EssentialsX/Essentials/issues/new/choose

2.  If you are fixing a performance issue, please use the "Bug fix" PR template
      instead. This is better suited to performance issues.

3.  Include a demonstration.
      If you are adding commands, please provide screenshots and/or a video
      demonstration of the feature. Similarly, if you are adding new API,
      please include a link to example code that takes advantage of your
      proposed API. This will aid us in reviewing PRs and speed up the process
      significantly.

-->

### Information

<!--
    Replace #nnnn with the original issue. If this PR implements features from
    multiple issues, you should repeat the "fixes #nnnn" for each issue. 

> -->

Reviewer: apply changes from above

This PR closes #nnnn. 

### Details

**Proposed feature**  
<!-- Type a description of your proposed feature below this line. -->

**Environments tested**
<!--
    Below this line, put an "x" inside the box for the environments you have
    tested this bug fix on, and if relevant alter the OS and Java version
    accordingly. If this feature does not apply to an environment, strike
    through the environment using ~~strikethrough~~. If you have tested on
    other environments, add a new line with relevant details.
-->

- [ ] Paper 1.16.3 (any OS, any Java 8+ version)
- [ ] CraftBukkit/Spigot/Paper 1.12.2 (any OS, any Java 8+ version)

> - [ ] CraftBukkit 1.8.8 (any OS, any Java 8+ version)

Reviewer: changes from above

---
name: Feature addition
about: Use this if your PR adds a new feature to EssentialsX.
labels: 'type: enhancement'
---

<!-- EssentialsX feature submission guide

NOTE: Failure to fill out this template properly may result in your PR being
      delayed or ignored without warning.

Don't put anything inside this block, as it will be hidden when you post your
PR. Make sure to also not type between any arrows in the template, as this text

> will also be hidden.

Reviewer 2: Same comment.

Want to discuss your PR before submitting it? Join the EssentialsX Development
server: https://discord.gg/CUN7qVb

If you are submitting a new feature, please follow the following steps:

1.  Fill out the template in full.
      This includes providing screenshots and a link to the original feature 
      request. If there isn't an existing feature request, we strongly
      recommend opening a new feature request BEFORE opening your PR to
      implement it, as this allows us to review whether we're likely to accept
      your feature in advance, and also allows us to discuss possible
      implementations for the feature. If there is no associated feature
      request, your PR may be delayed or rejected without warning.
      
      You can open a new feature request by following this link:
      https://github.com/EssentialsX/Essentials/issues/new/choose

2.  If you are fixing a performance issue, please use the "Bug fix" PR template
      instead. This is better suited to performance issues.

3.  Include a demonstration.
      If you are adding commands, please provide screenshots and/or a video

>       demonstration of the feature. Similarly, if you are adding new API,

Reviewer 3: ```suggestion
      demonstration of the feature. Similarly, if you are adding a new API,
```

Reviewer 2: Without 'a' reads better to me

>       please include a link to example code that takes advantage of your

Reviewer 3: ```suggestion
      please include a link to some example code that takes advantage of your
```

      proposed API. This will aid us in reviewing PRs and speed up the process
      significantly.

-->

### Information

<!--

>     Replace #nnnn with the number of the original issue. If this PR implements

Reviewer 3: ```suggestion
    Replace #nnnn with the number of the original issue. If this PR is implementing
```

Author: I don't feel this reads any better than the existing text

Reviewer 2: I think I like the original as well

    features from multiple issues, you should repeat the "closes #nnnn" for
    each issue. 
-->

This PR closes #nnnn. 

### Details

**Proposed feature**  
<!-- Type a description of your proposed feature below this line. -->

**Environments tested**
<!--
    Below this line, put an "x" inside the box for the environments you have
    tested this bug fix on, and if relevant alter the OS and Java version
    accordingly. If this feature does not apply to an environment, strike
    through the environment using ~~strikethrough~~. If you have tested on
    other environments, add a new line with relevant details.
-->

- [ ] Paper 1.16.3 (any OS, any Java 8+ version)
- [ ] CraftBukkit/Spigot/Paper 1.12.2 (any OS, any Java 8+ version)
- [ ] CraftBukkit 1.8.8 (any OS, any Java 8+ version)

**Demonstration**
<!--
    Below this block, include screenshots/code snippets from before and after
    as necessary. If you have created or used a test case plugin, please link

>     to a download of the plugin, source code and exact version used where

Reviewer 3: Same situation as above. If you decide to add, add it here too.
